tests/phpunit/CRM/Activity/BAO/ActivityTest.php

```diff
@@ -1154,4 +1154,140 @@ public function testSendSMSWithoutPermission() {
+
+    $outBoundSmsActivityId = CRM_Core_PseudoConstant::getKey('CRM_Activity_BAO_Activity', 'activity_type_id', 'SMS');
+    $activityStatusCompleted = CRM_Core_PseudoConstant::getKey('CRM_Activity_BAO_Activity', 'activity_status_id', 'Completed');
+    $details = 'createSendSmsTest text';
+    $this->assertEquals($activity['activity_type_id'], $outBoundSmsActivityId, 'Wrong activity type is set.');
+    $this->assertEquals($activity['status_id'], $activityStatusCompleted, 'Expected activity status Completed.');
+    $this->assertEquals($activity['subject'], 'createSendSmsTest subject', 'Activity subject does not match.');
+    $this->assertEquals($activity['details'], $details, 'Activity details does not match.');
+    $this->assertEquals(TRUE, $sent, "Expected sent should be true");
+    $this->assertEquals(1, $success, "Expected success to be 1");
+  }
+
+  /**
+   * @param int $phoneType (0=no phone, phone_type option group (1=fixed, 2=mobile)
+   */
+  public function createSendSmsTest($phoneType = 0) {
+    require_once 'CiviTest/CiviTestSMSProvider.php';
+    // Create an SMS provider "CiviTestSMSProvider". Civi handles "CiviTestSMSProvider" as a special case and allows it to be instantiated
+    //  in CRM/Sms/Provider.php even though it is not an extension.
+    $groupID = CRM_Core_DAO::getFieldValue('CRM_Core_DAO_OptionGroup', 'sms_provider_name', 'id', 'name');
+    $params = array(
+      'option_group_id' => $groupID,
+      'label' => 'unittestSMS',
+      'value' => 'unit.test.sms',
+      'name'  => 'CiviTestSMSProvider',
+      'is_default' => 1,
+      'is_active'  => 1,
+      'version'    => 3,
+    );
+    $providerValue = civicrm_api3('option_value', 'create', $params);
+    $provider = civicrm_api3('SmsProvider', 'create', array(
+      'name' => "CiviTestSMSProvider",
+      'api_type' => "1",
+      "username" => "1",
+      "password" => "1",
+      "api_type" => "1",
+      "api_url" => "1",
+      "api_params" => "a=1",
+      "is_default" => "1",
+      "is_active" => "1",
+      "domain_id" => "1",
+    ));
+    $smsProviderParams['provider_id'] = $provider['id'];
+
+    // Create a contact
+    $contactId = $this->individualCreate();
+    $contactsResult = $this->civicrm_api('contact', 'get', array('id' => $contactId, 'version' => $this->_apiversion));
+    $contactDetails = $contactsResult['values'];
+
+    // Get contactIds from contact details
+    foreach ($contactDetails as $contact) {
+      $contactIds[] = $contact['contact_id'];
+    }
+
+    $activityParams['sms_text_message'] = __FUNCTION__ . ' text';
+    $activityParams['activity_subject'] = __FUNCTION__ . ' subject';
+
+    // ActivityParams is overwritten by sendSms but we need it for results
+    $activityParamsCopy = $activityParams;
+
+    // Get a "logged in" user to set as source of Sms.
+    $session = CRM_Core_Session::singleton();
+    $sourceContactId = $session->get('userID');
+
```

Comment:
@mattwire you could here add 

```
     $config = &CRM_Core_Config::singleton();
     $config->userPermissionClass->permissions = array('access CiviCRM', 'send SMS');
```

Reply:
or, better 

CRM_Core_Config::singleton()->userPermissionClass->permissions = array('access CiviCRM', 'send SMS');

(that &CRM.... is C & P from somewhere)